homeassistant/components/automation/device.py

```diff
@@ -18,9 +18,6 @@
     platform = await async_get_device_automation_platform(
         hass, config[CONF_DOMAIN], "trigger"
     )
-    if hasattr(platform, "async_validate_trigger_config"):
```

Comment:
Why would we remove it from everything and not just ZHA ?

Reply:
I just wanted to quickly back this out since it broke ZHA on dev.
If you think the concept is still sane, we can just revert the ZHA part instead?

Reply:
I suggest we revert the ZHA part only. And actually, I think that we should have ZHA be conditional. If it is loaded, do a full validation, otherwise just check the schema.

Reply:
Done!